src/sentry/static/sentry/app/views/organizationContext.jsx

```diff
@@ -153,11 +166,23 @@ const OrganizationContext = createReactClass({
     fetchOrganizationDetails(
       this.props.api,
       this.getOrganizationSlug(),
-      this.props.detailed
+      this.props.detailed,
+      true // silent
     );
-    // create a request for all teams if in lightweight org
+    // create a request for all teams and all projects if in lightweight org
     if (!this.props.detailed) {
-      const teams = await this.props.api.requestPromise(
+      // create a new client so when this context unmounts the requests continue
```

Comment:
What do you think about moving all of this into the `fetchOrganizationDetails` action creator?

Reply:
That sounds good to me it also means we will have control of when to fetch projects and teams (during or after lightweight org). Should we have it so that those projects/ teams/ requests are blocked until after lightweight org comes back with a response?